### DISCUSSION TITLE
git-lfs on raspberry pi

### DISCUSSION THREAD

**Discussion by smathema** [git-lfs/git-lfs#5150] [Q&A] [answered]
I've been trying to install git-lfs on Raspberry Pi 3 Model B Plus Rev 1.3, followed a few articles but was unable to get it to work, please let me know if there is a workaround to get it on the Pi.

**Comment by bk2204** [comment, marked as answer]
Hey,

That machine uses an arm64 processor, so you can download the Linux ARM64 release asset from https://github.com/git-lfs/git-lfs/releases/tag/v3.2.0 and that should work on your system.  The binary is relocatable so you can move it wherever you like in the `PATH`.

**Reply by smathema** [reply]
Still no luck using that as well:
```
$ ls git-lfs-linux-arm64-v3.2.0.tar.gz 
git-lfs-linux-arm64-v3.2.0.tar.gz

$ ls -al /usr/local/bin/git-lfs
-rwxr-xr-x 1 root staff 10354688 Oct 28 12:04 /usr/local/bin/git-lfs

$ git lfs install
/usr/local/bin/git-lfs: 1: /usr/local/bin/git-lfs: ELF��H@�@8@@@@�����dd��H��HIJJ�M�M���: not found
/usr/local/bin/git-lfs: 2: /usr/local/bin/git-lfs: /usr/local/bin/git-lfs: 2: /usr/local/bin/git-lfs: LjLi�: not foundQ�t�e��HjJI: not found

/usr/local/bin/git-lfs: 3: /usr/local/bin/git-lfs: �: not found
/usr/local/bin/git-lfs: 4: /usr/local/bin/git-lfs: ��: not found
/usr/local/bin/git-lfs: 1: /usr/local/bin/git-lfs: Syntax error: Unterminated quoted string
```

OS info:
```
$ cat /etc/os-release 
PRETTY_NAME="Raspbian GNU/Linux 9 (stretch)"
NAME="Raspbian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
```

**Reply by bk2204** [reply]
Can you provide the output of `uname -a`, please?

**Reply by smathema** [reply]
```
$ uname -a
Linux raspberry 4.14.62-v7+ #1134 SMP Tue Aug 14 17:10:10 BST 2018 armv7l GNU/Linux
```

**Reply by bk2204** [reply]
Okay, then you want the Linux ARM (not Linux ARM64) release, since you're using a 32-bit kernel.  That should work, since it's an ELF executable.  I would expect that you wouldn't see the message you're seeing and instead see an "Exec format error" for the ARM64 binary, but if you continue to see problems with the Linux ARM binary, then you'd need to download your own Go binaries and compile your own Git LFS binary as outlined in the README.

**Reply by smathema** [reply]
Thanks, it worked!!
